src/job/jobHandler.ts

```diff
@@ -504,12 +544,65 @@ export abstract class JobHandler {
     return await this.executeBuild();
   }
 
+  /* New build process without Makefiles */
+  @throwIfJobInterupted()
+  protected async build(): Promise<boolean> {
+    this.cleanup();
+    const job = this._currJob;
+    const logger = (msg: string) => this.logger.save(this.currJob._id, msg);
+
+    await this.cloneRepo(this._config.get<string>('repo_dir'));
+    this._logger.save(job._id, 'Cloned Repo');
+    await this.commitCheck();
+    this._logger.save(job._id, 'Checked Commit');
+    await this.pullRepo();
+    this._logger.save(job._id, 'Pulled Repo');
+    await this.setEnvironmentVariables();
+    this.logger.save(job._id, 'Prepared Environment variables');
+    const buildDependencies = await this.getBuildDependencies();
```

Comment:
Since we aren't using the makefiles, we don't need to do everything here and before.

Reply:
Sorry, I was getting all mixed up. I was working on another branch where I added the clone functionality to the [`getBuildDependencies` function](https://github.com/mongodb/docs-worker-pool/blob/32ae0a7c709f8a3071b4cec36c20c1a020acbaa1/src/commands/src/helpers/dependency-helpers.ts#L18). 

Looking at the commitCheck more closely, it looks like we do need something to do that logic if we were to remove it. It looks like it [checks out the branch and make sure the commit exists](https://github.com/mongodb/docs-worker-pool/blob/040d812d52b4be5bfcd978f6be7a9efc78fed76b/src/services/commandExecutor.ts#L92).

Now that I've thought about it a bit more, I'll leave it up to you if you want to keep this!